app/models/concerns/account_settings.rb

```diff
@@ -136,6 +137,12 @@ def set_type(value, to_type)
       value.is_a?(String) ? JSON.parse(value) : value
     when 'string'
       value.to_s
+    when 'json_editor'
+      begin
+        JSON.pretty_generate(JSON.parse(value))
+      rescue JSON::ParserError
+        value
```

Comment:
I am relatively sure `pretty_generate` or `parse` do not mutate the value so if that is true may I suggest setting it to a variable and returning that?  I could be misunderstanding the `begin` and `rescue` block so if that's true then feel free to ignore this.

Should the json be cleaned of any `"_proto_"` references before being returned?  Not sure if that is overkill for here or if that is done elsewhere.  Found some JS that does this.

```
const cleaner = (key, value) => key === "__proto__" ? undefined : value;

JSON.parse(json, cleaner)

```

Reply:
You're correct that neither `parse` nor `pretty_generate` mutate the value. If you call `JSON.parse` on a string that is not valid JSON, it throws a `JSON::ParserError`. Since we're dealing with a value that comes from user input, we can't trust it to always be valid JSON. Because of that, we plan for a `JSON::ParserError` to be thrown and, when it is, we `rescue` and simply give the user back what they entered so they can correct it. 

[This recording](https://share.zight.com/jku090b2) shows what I'm trying to describe. This method simply returns `value` if it's invalid JSON, which is how it shows up in the text box (otherwise the text box would be empty if we returned nothing, forcing the user to start over from scratch). `#validate_json` is what handles adding the error message if `value` is not valid JSON. 

> Should the json be cleaned of any `"_proto_"` references before being returned? ...

I don't think so. I'm honestly not familiar with `_proto_` in the context of JSON. Since this JSON specifically is meant to represent Bulkrax field mappings (which don't use `_proto_`), I don't think we need to plan for that possibility

Reply:
Ahh, that makes much more sense.  I had a feeling I was missing something about that rescue block.  Appreciate the explanation.

For the `"_proto_"` I was just doing some JSON sanitizing best practices and stumbled upon a [stackoverflow article](https://stackoverflow.com/questions/63926663/how-should-untrusted-json-be-sanitized-before-using-json-parse) about preventing a regular user from using that syntax to make themselves an admin like the code snippet below.  However, I absolutely understand if that is not a rational concern given the way that JSON is used.

```
json = '{"user":{"__proto__":{"admin": true}}}';
```